From: Giridhar Malavali <gmalavali@marvell.com>
Date: Thu, 24 Jan 2019 23:23:46 -0800
Subject: scsi: qla2xxx: Fix SRB allocation flag to avoid sleeping in IRQ
 context
Patch-mainline: v5.1-rc1
Git-commit: 97a93cea887376e13ea7d473ab349181850f4e39
References: bsc#1123034 bsc#1131304 bsc#1127988

This patch fixes SRB allocation flag from GFP_KERNEL to GFP_ATOMIC, to
prevent sleeping in IRQ context

Signed-off-by: Giridhar Malavali <gmalavali@marvell.com>
Signed-off-by: Himanshu Madhani <hmadhani@marvell.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/scsi/qla2xxx/qla_init.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/scsi/qla2xxx/qla_init.c
+++ b/drivers/scsi/qla2xxx/qla_init.c
@@ -1831,7 +1831,7 @@ qla24xx_async_abort_cmd(srb_t *cmd_sp, b
 	int rval = QLA_FUNCTION_FAILED;
 
 	sp = qla2xxx_get_qpair_sp(cmd_sp->vha, cmd_sp->qpair, cmd_sp->fcport,
-	    GFP_KERNEL);
+	    GFP_ATOMIC);
 	if (!sp)
 		goto done;
 
